assets/js/src/form_editor/store/actions.ts

```diff
@@ -33,15 +33,21 @@ export function* toggleFullscreen(toggleTo) {
   };
 }
 
-export function changeFormBlocks(blocks) {
-  return {
+export function* changeFormBlocks(blocks) {
+  yield {
+    type: 'CREATE_HISTORY_RECORD',
```

Comment:
Why do we need a new action? This is called when other actions are also triggered. I don't see why is this necessary. We can have this in the reducer, I think that would be nicer and keep the actions cleaner

Reply:
It isn't necessary. I preferred this solution. I think that this functionality will be in reducers a little bit hidden. But if you want to move it I have no problem with it.

Reply:
I think putting it in reducer would be nicer. We have one big reducer for all actions, but we can create another reducer only for undo/redo it would listen to all actions that need creating history. We can use [combineReducers](https://developer.wordpress.org/block-editor/packages/packages-data/#combineReducers) to combine the reducer we have now and this undo-redo. I think this is the recommended approach.